src/dune_engine/rule.ml

```diff
@@ -36,7 +36,7 @@ module Promote = struct
   type t =
     { lifetime : Lifetime.t
     ; into : Into.t option
-    ; only : Predicate_lang.Glob.t option
+    ; only : string Predicate.t option
```

Comment:
This seems like an unfortunate diff: `Predicate_lang.Glob.t` is much clearer to me than `string Predicate.t`. Can we keep the `Glob` module?

Reply:
Funny you say that because the `Predicate_lang.Glob.t` is a totally misleading name. It has nothing to do with globbing and it in fact only consists with raw predicates `string -> bool` :)

But I'd argue that even if the name was accurate, I think it's better for the build system to not care about the _how_ of predicates were constructed unless it's needed for something concrete. The predicate language is a part of the dune frontend - the language, and is subject to evolution. It should be decoupled from the engine as much as possible.

Reply:
Sure, I understand that `Predicate_lang.Glob.t` is nothing more than `string -> bool` conceptually. It's the `string` that the problem here: when I see `only : string Predicate.t option` I have no idea what strings we are talking about. If it was something like `Filename.t Predicate.t` if would be fine. (`Predicate_lang.Glob.t` solves this problem by giving an alias to a different thing: instead of aliasing `string`, it aliases the whole predicate, which does clear things up.)

Reply:
I like the `Filename.t Predicate.t` idea. At one point, I lobbied to introduce a `Path.Component.t` type which would be just a string without any `/` characters, but it didn't go through. What invariants do you have in mind for `Filename.t`? Doesn't have `/` and must not be `.` or `..`?

Reply:
> What invariants do you have in mind for `Filename.t`? Doesn't have `/` and must not be `.` or `..`?

Yes, that sounds right. I'd love to have a type for this!

Reply:
Sure, will do that. Best to leave it to a separate PR though.

Reply:
Can we keep the old (more self-explanatory) type alias until we have `Filename.t`? Or at least add a comment?

Reply:
I added a commit that introduces `Filename.t` as an alias. Will use it more systematically in the rest of the codebase in a subsequent PR.